### ISSUE TITLE
Advanced features not working (create new mod, compare to original Witcher script)

### ISSUE THREAD

**Issue by TheBloke** [nicollasricas/vscode-witcherscript#5]
Hello,

Firstly, thanks so much for creating this extension.  I just started investigating modding TW3, and your extension has already been incredibly helpful. I had never used VSCode before, as I am a long-time Vim user.  But seeing your extension was available I tried out VSCode, which has an excellent Vim plugin, and it has made it a lot easier to navigate and understand Witcher 3 vanilla code, and code added by the mods I use.  I couldn't live without the syntax highlighting now.

I was hoping to use some of the more advanced functions of your extension, like making a new mod, the comparison feature to vanilla scripts, etc, but unfortunately I can't seem to get these to work.

I'm wondering if maybe something changed in later VSCode releases, and your extension isn't working fully any more?

As an example, I just tried to Make New Package.  I entered a name (modTestVS1) and specified an empty destination folder.  This is the result:
![image](https://user-images.githubusercontent.com/784313/80925692-db72ef00-8d89-11ea-87a8-788ea8075aeb.png)

You can see that it's tried to add the folder to a new Workspace, but the folder is invalid.  Specifically, it's missing the drive letter (should be `C:\Users\tomjo\Documents...`), so the folder shows as invalid.

Similarly I've not had any luck getting Compare With Original Witcher Script to do anything.  I've tried opening a random .ws file from a mod, then running this command, but nothing happens.

My extension settings are as follows:
![image](https://user-images.githubusercontent.com/784313/80925739-3278c400-8d8a-11ea-8e60-3cebd95ec8e7.png)

I've not yet configured an uncooked base path, as I've not uncooked my game yet.  I assumed this wouldn't be required for script work, as all the vanilla scripts can be found under the Game path, but maybe that's wrong?

I realise you may not be working on this extension or Witcher 3 modding any more, so don't worry if you're no longer interested in supporting this.  I can investigate TS script myself and see maybe if I can understand what's going wrong.

But I thought I'd try sending a support message in case you know of any quick solutions to these issues, or could point me in the direction of what I should check first in the code.

Thanks again for all your work on this.

**Comment by nicollasricas** [comment]
I have stopped for a time, but I came back after seen that most of the mods I'm playing with are still in development. I'm working on a final version that will be released this week.

I will keep you in the loop.

**Comment by TheBloke** [comment]
Thanks so much for the fast reply, and it's wonderful to hear you're still working on it.

I noticed in the description that it listed "Navigation menu, symbols and references" as WIP.  Does that mean that, once it's finished, the VSCode Outline view will work with .ws files?  

If so that would be excellent; that's probably the #1 feature I wish I had.  Working through 15,000 line files it would be so helpful to be able to quickly scroll through a list that showed every class, function, enum, etc, and be able to click on any one to jump straight to it.

Thanks again for the response and all the work you're doing.  Can't want to try the final version!

**Comment by nicollasricas** [comment]
> I noticed in the description that it listed "Navigation menu, symbols and references" as WIP. Does that mean that, once it's finished, the VSCode Outline view will work with .ws files?

Yes, just the outline at first. I plan to add other stuff like definitions, hovering, but for that, I need a parser which is not that hard to do it but takes time.

I will do it eventually because it's been fun to learn to do that.

**Comment by TheBloke** [comment]
Hi Nicollas,

Hope everything is going well.  I've been working on mod code all week in VSCode and your extension has been invaluable.

Now I have a bit more experience with WS, and your extension, I thought I'd let you know, in case you didn't already, of a couple of minor syntax highlighting issues.

Firstly, I've found one type of syntax that causes highlighting to break.  This unusual syntax:
`var VarName : Type = Function();` will stop syntax highlighting until the next full `var` line.

Here's an example:
![image](https://user-images.githubusercontent.com/784313/82033465-aa04f800-9694-11ea-9c0e-171220798529.png)

I also noted a few instances where extra syntax highlighting could occur: 

1. In a state definition, the class on which the state is defined is currently not highlighted (but the parent class is):
![image](https://user-images.githubusercontent.com/784313/82035306-40d2b400-9697-11ea-9fc8-a49ce0b656b7.png)

2. `virtual_parent` is an alternative syntax for `parent`, as in this example:
![image](https://user-images.githubusercontent.com/784313/82033819-30b9d500-9695-11ea-825c-330edec2b5f8.png)

3. `cleanup function` is a rare function type used in a `state`:
![image](https://user-images.githubusercontent.com/784313/82033877-4929ef80-9695-11ea-8b27-c964f65fb804.png)

These are all minor issues, but I thought I'd mention them while you're working on the extension.

Thanks very much!

**Comment by nicollasricas** [comment]
Hey,

Thanks for reporting, some of those was already fixed.

I will make sure to fix them all.

It will be released soon.

**Comment by TheBloke** [comment]
Thanks so much!

**Comment by nicollasricas** [comment]
![image](https://user-images.githubusercontent.com/7860985/83342258-43e6ba80-a2bb-11ea-8df9-ac1c9c20026c.png)

Almost done, the outline feature was done with a workaround until I finish the parser, but it's working pretty great.

**Comment by TheBloke** [comment]
That's looking really great!  Love it.  Can't wait to try it, it'll definitely make development a lot easier.

Thanks for all your work on this.

**Comment by eshate** [comment]
Hey! I'm somewhat learning code on modding, so excuse me. I'm not making mods - just want to have better insight when merging - that's why I am happy to see your work on extension for VSCode. But still, I have some problems. I saw that there was a Wiki - but no more. Two things:

1. Can you upload a Wiki somewhere?
2. Do I need a ModKit for extension to work properly? 

Thanks in advance!

And also: I've noticed that I have very outdated version. From where I can download the 2.0.9?
